asn1ate/sema.py

```diff
@@ -276,9 +278,17 @@ def resolve_type_decl(self, type_decl):
         declaration.
         """
         user_types = self.user_types()
-
+        
         if isinstance(type_decl, ReferencedType):
-            return self.resolve_type_decl(user_types[type_decl.type_name])
+            if type_decl.module_name is None or type_decl.module_name == self.name:
```

Comment:
You can simplify this by first finding the module, and then calling resolve_type_decl, e.g.

    module = None
    if type_decl.module_name...:
        module = self
    else:
        module = ...

    if not module:
        raise Exception(...)

    return module.resolve_type_decl(...)